ui/src/app/components/admin/audit-card-list-view.tsx

```diff
@@ -193,17 +210,115 @@ const AuditActionCard = (props: { action: AuditAction }) => {
+        </div>
+    </React.Fragment>);
+};
+
+export const AuditActionCardListView = (props: { actions:  AuditAction[]}) => {
   const {actions} = props;
+  const [actionTypes, setActionTypes] = useState([]);
+  const [activeActionTypes, setActiveActionTypes] = useState({});
+  const [actionTypeToCardCount, setActionTypeToCardCount] = useState({});
+
+  useEffect(() => {
+    const result = (fp.flow(
+      fp.flatMap((action: AuditAction) => action.eventBundles),
+      fp.map((eventBundle) => eventBundle.header.actionType),
+      fp.sortBy(a => a),
+      fp.sortedUniq)(actions));
+    setActionTypes(result);
+  }, [actions]);
+
+  useEffect(() => {
+    const actionTypeToShow: Object = {};
+    fp.forEach((propertyPath: string) => {
+      // TODO: figure out why I can't make this work using fp.set()
+      actionTypeToShow[propertyPath] = {
+        isActive: true,
+        displayName: `${toTitleCase(propertyPath)} (${actionTypeToCardCount[propertyPath] || 0})`
+      };
+    })(actionTypes);
+    setActiveActionTypes(actionTypeToShow);
+  }, [actionTypes]);
+
+  useEffect(() => {
+    const result: { [key: string]: number } = {};
+    fp.forEach(action => {
+      fp.forEach((actionType: string) => {
+        result[actionType] = (result[actionType] || 0) + 1;
+      })(getActionTypes(action));
+      setActionTypeToCardCount(result);
+    })(actions);
+  }, [actions]);
+
+  const updateFilterCallback = (actionType: string, isActive: boolean) => {
+    const newActiveActionTypes = fp.clone(activeActionTypes);
+    newActiveActionTypes[actionType].isActive = isActive;
+    setActiveActionTypes(newActiveActionTypes);
+  };
+
+  function getActionTypes(action: AuditAction) {
+    return fp.map((e: AuditEventBundle) => e.header.actionType)(action.eventBundles);
+  }
 
-  // Temporary workaround for sort order in the APIs, fixed in RW-4999.
-  const actionsSorted = actions.sort((a, b) => {
-    return new Date(b.actionTime).getTime() - new Date(a.actionTime).getTime();
-  });
+  // An action card should be shown only iff all action types
+  // in its bundles are active (checked). Arguably this could be a simple boolean
+  // property on the card, but then we'd have to manage that from a parent component
+  // and use a callback, so it's not a big win.
+  const shouldShowAction = (action: AuditAction) => {
+    const cardActionTypes = getActionTypes(action);
+    return fp.all((actionType: string) =>
+        fp.get(`${actionType}.isActive`)(activeActionTypes))(cardActionTypes);
+  };
 
   return (
-      <div style={{margin: '1rem', width: '30rem'}}>
-        {actionsSorted.map((action, index) => (<AuditActionCard key={index} action={action}/>))}
-      </div>
+      <React.Fragment>
+        <ActionTypeFilter activeActionTypes={activeActionTypes}
+                          updateFilter={updateFilterCallback}
+        />
+        <div style={{margin: '1rem', width: '30rem'}}>
+          {fp.any(shouldShowAction)(actions)
```

Comment:
I know we differ in opinion on who should be responsible for conditional renders (parent or component).
After searching the code base I only saw 2-3 instances of the `show` property, but many of the `&& <` string (I believe that accurately indicates a conditional render)

Here is a suggestion that may simplify things a bit / improve readability:

```javascript
  const filteredActions = fp.flow(
    fp.toPairs,
    fp.map(([index, action]) => shouldShowAction(action) && <AuditActionCard key={index} action={action}/>),
    fp.without([false])
  )(actions);

  return (
      <React.Fragment>
        <ActionTypeFilter activeActionTypes={activeActionTypes}
                          updateFilter={updateFilterCallback}
        />
        <div style={{margin: '1rem', width: '30rem'}}>
          {filteredActions.length ? filteredActions : <div  style={infoMessageStyle}>All cards filtered out.</div>}
        </div>
      </React.Fragment>
  );
```

Followed by a removal of the show prop in the `AuditActionCard` component